modules/ROOT/pages/fd-iterating.adoc

```diff
@@ -0,0 +1,112 @@
+
+The Parallel For Each component does not iterate over the elements in a collection. Therefore, the value of a variable that enters the Parallel For-Each scope is the same at the start of the processing of each variable. New variables or modifications of already existing variables during the processing of one element are not visible during the processing of another element. Moreover, no changes to variables with the scope of a Parellel For-Each component are available to the subsequent cards in the flow.
+
+Suppose that, as in the previous example, you use a Set Variable component just before the scope of a Parallel For-Each component. Within the Set Variable component, you set the variable `NumOfElements` to 0. You then use a Set Variable component in the Parallel For-Each scope, using the same variable, that specifies a DataWeave script that increments the value by 1 every time it is called. For each parallel process that the component runs, the highest value reached by `NumOfElements` is 1. And if subsequent cards in the flow use `NumOfElements`, its initial value after the Parallel For-Each component finishes processing the collection is 0, which is the same value it had before the Parallel For-Each component started.
+
+
+== Error Handling
+
+Both the For-Each component and Parallel For-Each component stop processing a collection if an element causes any component, connector, or module in a scope to throw an exception. In addition, the Parallel For-Each component stops processing a collection if one of the parallel processes times out while waiting for an open connection to a system. You can set the maximum length of time that a parallel process can wait.
+
+When the processing of a collection stops because of an exception or timeout, Flow Designer displays a red circle on the card that represents the component in the flow. You can open the *Logs* pane to read the error message.
+
+== Configuring a For-Each Component
+
+When you configure this component, you must specify values in these fields:
+
+Collection:: An expression that returns a Java collection, object array, map, or DOM nodes.
+
+Batch Size:: Partitions the collection into sub-collections of the specified number of elements. For example, if a collection has 200 elements and you set the batch size to 50 elements, the component processes 4 batches of 50 elements each.
+
+Root Message Variable Name:: Name of the property that stores the parent message. The parent is the complete, non-split message. // What is the root message variable name for? What does it store "the parent message" for? Is "parent message" the full collection?
+
+Counter Variable Name:: Name of the property that stores the number of messages over which it iterates. // What is "it"?
```

Comment:
The counter variable name is the name of the variable that will be set to be used inside the for each scope to identify the index of the element of the collection being processed. That is, if the for each receives as collection like [1, 2, 3, 4, 5] and it's processing the second element (2), the variable's value will be 1 (because it has already processed the element 0 and it's now processing the element 1).